MIGRATION.ru.md

```diff
@@ -1,5 +1,813 @@
+        }
+    }
+});
+```
+
+#### БЭМ-события на экземплярах
+
+Было:
+
+```js
+BEMDOM.decl('my-block', {
+    onSetMod : {
+        'js' : {
+            'inited' : function() {
+                this.findBlockOutside('my-block-2').on('my-event', this._onMyBlock2MyEvent, this);
+            },
+
+            '' : function() {
+                this.findBlockOutside('my-block-2').un('my-event', this._onMyBlock2MyEvent, this);
+            }
+        }
+    }
+});
+```
+
+Стало:
+
+```js
+bemDom.declBlock('my-block', {
+    onSetMod : {
+        'js' : {
+            'inited' : function() {
+                this._events(this.findParentBlock('my-block-2')).on('my-event', this._onMyBlock2MyEvent);
+            }
+        }
+    }
+});
+```
+
+#### Делегированные DOM-события
+
+Было:
+
+```js
+BEMDOM.decl('my-block', { /* ... */ }, {
+    live : function() {
+        this.liveBindTo('click', this.prototype._onClick);
+    }
+});
+```
+
+Стало:
+
+```js
+bemDom.declBlock('my-block', { /* ... */ }, {
+    onInit : function() {
+        this._domEvents().on('click', this.prototype._onClick);
+    }
+});
+```
+
+Было:
+
+```js
+BEMDOM.decl('my-block', { /* ... */ }, {
+    live : function() {
+        this.liveBindTo('my-elem', 'click', this.prototype._onMyElemClick);
+    }
+});
+```
+
+Стало:
+
+```js
+bemDom.declBlock('my-block', { /* ... */ }, {
+    onInit : function() {
+        this._domEvents('my-elem').on('click', this.prototype._onMyElemClick);
+    }
+});
+```
+
+#### Делегированные БЭМ-события
+
+Было:
+
+```js
+BEMDOM.decl('my-block', { /* ... */ }, {
+    live : function() {
+        this.liveInitOnBlockInsideEvent('my-event', 'my-block-2', this.prototype._onMyBlock2MyEvent);
+    }
+});
+```
+
+Стало:
+
+```js
+bemDom.declBlock('my-block', { /* ... */ }, {
+    onInit : function() {
+        this._events(MyBlock2).on('my-event', this.prototype._onMyBlock2MyEvent);
+    }
+});
+```
+
+Следует обратить внимание, что параметр с функцией обработчиком события теперь обязательный.
+
+Было:
+
+```js
+modules.define('my-block', ['i-bem__dom', 'my-block-2'], function(provide, BEMDOM) {
+
+provide(BEMDOM.decl(this.name, { /* ... */ }, {
+    live : function() {
+        this.liveInitOnBlockInsideEvent('my-event', 'my-block-2');
+    }
+}));
+
+});
+```
+
+Стало:
+
+```js
+modules.define('my-block', ['i-bem-dom', 'my-block-2', 'functions'], function(provide, bemDom, MyBlock2, functions) {
+
+provide(bemDom.declBlock(this.name, { /* ... */ }, {
+    onInit : function() {
+        this._events(MyBlock2).on('my-event', functions.noop);
+    }
+}));
+
+});
+```
+
+Было:
+
+```js
+BEMDOM.decl('my-block', {
+    onSetMod : {
+        'js' : {
+            'inited' : function() {
+                MyBlock2.on(this.domElem, 'my-event', this._onMyBlock2MyEvent, this);
+            },
+
+            '' : function() {
+                MyBlock2.un(this.domElem, 'my-event', this._onMyBlock2MyEvent, this);
+            }
+        }
+    }
+});
+```
+
+Стало:
```

Comment:
not sure which line you writing about, but since we have events API with participation of both blocks (block with event on and block which do binding) we can automatically unbind all regular cases